### DISCUSSION TITLE
password encryption type

### DISCUSSION THREAD

**Discussion by ssjay18** [nextgenhealthcare/connect#4903] [Q&A] [answered]
After creating a channel will postgres db access mirth will auto create some tables in which there is a table named persons password which stores the users login password and that password is in encrypted format i've seen database passwords will be stored using the base64 function to encode and decode the passwords but here there is some other encryption type is used "Does anyone have idea what type of encryption is done?"

**Comment by pacmano1** [comment, marked as answer]
Please read the user manual.  https://www.nextgen.com/-/media/files/nextgen-connect/nextgen-connect-39-user-guide.pdf

Page 443 in that link.

**Reply by ssjay18** [reply]
in the page you mentioned there was only abt the thread and destination thing but im asking in postgres db there is a table called person password with holds the password of the mirth connect login credentials and that password is stored in a encrypted way and what encryption have they used to encrypt the password?

**Reply by pacmano1** [reply]
That page shows the encryption.   

![image](https://user-images.githubusercontent.com/44065187/146000974-34ddfed4-7a54-46ea-b211-2a8a04640bd1.png)

You can also look at the Mirth source code to see that.

**Reply by pacmano1** [reply]
The algorithm is AES, the encryption key length is 128, and the digest algorithm for creating salted hash valuess for user passwords is SAH256.

**Reply by ssjay18** [reply]
0thanks for the answer sorry the mistake was mine I didnt look the page no properly  thanks for the help :)

**Comment by ssjay18** [comment]
thanks for the answer 👍  
the answer for this is in page 443 
https://www.nextgen.com/-/media/files/nextgen-connect/nextgen-connect-39-user-guide.pdf
